### PR TITLE
GTPv2: fix IE type value and add Port Number IE.

### PR DESCRIPTION
**Pull request by artisdom**
Fix IE type value and add Port Number IE.

### CODE REVIEW

**Comment by alexsult** [comment]
Hi @artisdom,

Nice for the IE type fix ! I don't know how we missed that !
For the new IE, I think that a more precise name could be nice (like  IE_Source_Port for example). 
Finally, could you add a test for the new field in "gtp_v2.uts" ?

Alexis

**Comment by artisdom** [comment]
Hi @alexsult, 

I re-check the 3GPP TS 29.274, and thinking that the "Port Number" IE is probably defined to carry any UDP/TCP port, not only for "Source Port". I've changed the IE name to IE_Port_Number to match with the standard, and two new tests are added for it in "gtp_v2.uts". Please have a look, thanks.

Ted

**Comment by alexsult** [comment]
Hi @artisdom 

Thanks for the tests !

Yes, there may be a doubt about this field, it's called in "Source port" in the title of the section 8.56 of the ts 29.274 (rev 13.6.0), but in the same section we find "UDP Source Port Number" in the figure's caption. Again, page 224 of this document, we find (in the table 8.1-1) the IE field 126 under "UDP Source Port Number". For me, this IE (126) can only be an UDP Source port. Could you tell me where I can find in the TS something which tells that it can be a TCP port or a destination port ?
I'm just asking that because I think that PortNumber isn't enough meaningful

Alexis

**Comment by artisdom** [comment]
Hi @alexsult,

There is update with [TS 29.274](http://www.3gpp.org/ftp//Specs/archive/29_series/29.274/), the latest one is [29274-f10.zip](http://www.3gpp.org/ftp/Specs/archive/29_series/29.274/29274-f10.zip).

In this newer version, in section 8.56, the figure's caption has been changed to "Figure 8.56-1: Port Number", also near the end of "Table 7.2.1-1: Information Elements in a Create Session Request"(page 50, 54), we can see the "Port Number" IE has been used in for both "UE UDP Port" and "UE TCP Port". Also there's no mention about whether it's the source or the destination. So I think the definition is more general now, it can be used to carry any type of port, albeit the current use may be only for UDP/TCP source port. 

Best Regards,
Ted

**Comment by alexsult** [comment]
Hi, 

Thanks, I haven't checked the latest version, my bad. So that's ok for me.

Best regards,

Alexis

File: scapy/contrib/gtp_v2.py [hunk 1]
```diff
@@ -390,6 +391,7 @@ class IE_Indication(gtp.IE_Base):
 
     ]
 
+
```
Comment (guedou) [inline scapy/contrib/gtp_v2.py]: Could you remove this unnecessary change ? Except this point, the PR looks good. Thanks!

Comment (artisdom) [reply scapy/contrib/gtp_v2.py]: Sure, updated.
